core/src/main/java/org/apache/iceberg/rest/RESTSerializers.java

```diff
@@ -110,6 +118,28 @@ public void serialize(
     }
   }
 
+  public static class UpdateReqDeserializer
```

Comment:
Maybe these classes should not be `public` so that we can change them? I think the only way `RESTSerializers` is currently used is through the `registerAll` method and that shouldn't require these to be public.

Reply:
good call, yes I think it makes sense to make these non-public